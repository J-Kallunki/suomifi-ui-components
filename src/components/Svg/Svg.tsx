import React from 'react';
import styled from '@emotion/styled';
import ReactSVG from 'react-svg';
import { ariaLabelOrHidden } from '../utils/aria';

export interface SvgProps {
  /** Image file */
  src: string;
  /** Fill-color */
  color?: string;
  /** Aria-label for SVG, undefined hides SVG from screen reader
   * @default undefined
   */
  ariaLabel?: string;
  labelName?: string;
  /** Custom classname to extend or customize */
  className?: string;
  /** Custom classname to SVG-element extend or customize */
  svgClassName?: string;
  /** Show mouse cursor as hand-pointer */
  pointer?: boolean;
  testId?: string;
}

export const Svg = styled(
  ({ color, labelName, ariaLabel, testId, pointer, ...props }: SvgProps) => {
    const fill = !!color ? color : '';
    const pointerStyle = !!pointer ? { cursor: 'pointer' } : {};
    return (
      <ReactSVG
        {...props}
        {...ariaLabelOrHidden(ariaLabel)}
        svgStyle={{
          fill,
          maxWidth: '100%',
          maxHeight: '100%',
          ...pointerStyle,
        }}
        role="img"
        data-testid={testId}
      />
    );
  },
)`
  label: ${({ labelName }) => (!!labelName ? labelName : 'svg')};
  max-width: 100%;
  ${({ pointer }) => !!pointer && ' * { cursor: pointer; }'}
`;
