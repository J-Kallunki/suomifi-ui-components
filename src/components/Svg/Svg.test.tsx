import React from 'react';
import { render } from 'react-testing-library';

import { Svg } from './Svg';
import testSvg from './test.svg';

test('calling render with the same component on the same container does not remount', () => {
  const svgRenderer = render(<Svg src={testSvg} testId="svg" />);
  const { getByTestId, container, rerender } = svgRenderer;
  expect(container.firstChild).toMatchSnapshot();
  expect(getByTestId('svg')).toBeDefined();

  // re-render the same component with different props
  rerender(<Svg src={testSvg} testId="gvs" />);
  expect(getByTestId('gvs')).toBeDefined();
});
