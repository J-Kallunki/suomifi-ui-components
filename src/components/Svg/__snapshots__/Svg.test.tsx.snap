// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calling render with the same component on the same container does not remount 1`] = `
<div
  aria-hidden="true"
  class="css-1f6zlyd-svg"
  data-testid="svg"
  role="img"
>
  <div>
    <div
      data-src="\\"test.svg\\""
      src=""
      style="max-width:100%;max-height:100%"
    />
  </div>
</div>
`;
